### ISSUE TITLE
Can't find WSL2 blackend option from the menu( Setting -> General )

### ISSUE THREAD

**Issue by wkissw** [docker/for-win#5244]
- [y] I have tried with the latest version of my channel (Stable or Edge)

### Expected behavior
I have enable WSL 2 feature on Windows. 
But I can't find the  WSL2 blackend option from the menu

my wsl list
```
PS C:\WINDOWS\system32> wsl -l -v
  NAME            STATE           VERSION
* kali-linux      Running         2
  Ubuntu-18.04    Running         2
```

### Actual behavior
Can't find WSL2 blackend option from the menu( Setting -> General )

### Information
  - Windows Version: 19033
  - Docker Desktop Version: 2.1.6.1
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: no

**Comment by mat007** [comment]
That’s what the General settings section is supposed to look like, can you post yours?

![image](https://user-images.githubusercontent.com/2386884/69845340-aad5d900-1270-11ea-825a-b11cf063bd69.png)

**Comment by wkissw** [comment]
> That’s what the General settings section is supposed to look like, can you post yours?
> 
> ![image](https://user-images.githubusercontent.com/2386884/69845340-aad5d900-1270-11ea-825a-b11cf063bd69.png)

I have tried many times before , but it looks like my machine is too slow.   When I woke up, this option appeared !!!  😮 
Amazing. . .

![image](https://user-images.githubusercontent.com/18184630/69846582-703f6100-12af-11ea-8256-7748fb8c7a98.png)

**Comment by mikeparker** [comment]
There was a rare race condition in detection of WSL 2 enabled environments which is now fixed with the latest edge release 2.1.7.0, you can download from here https://docs.docker.com/docker-for-windows/edge-release-notes/

I'm closing this as fixed - please let me know if you still have this issue with the new update and i'll reopen.

**Comment by LiamKarlMitchell** [comment]
@mikeparker Hey, I can't find the menu option?

2.2.0.0 and WSL 2 is installed/upgraded on Ubuntu.

![image](https://user-images.githubusercontent.com/1968109/73703598-4abe5000-4755-11ea-8842-8a46bf3d6a55.png)

![image](https://user-images.githubusercontent.com/1968109/73703617-69244b80-4755-11ea-9351-49379ee29ae9.png)

Docker Engine v19.03.5

**Comment by LiamKarlMitchell** [comment]
Okay, I closed completely out of Docker Desktop and re-ran it, waited for it to popup telling me it was ready to open console and get going, then when I opened settings it showed.

I guess it had not finished loading before I had opened the Settings menu...

**Comment by mikeparker** [comment]
@LiamKarlMitchell was this the first time you'd run the new version of Docker Desktop? 

Docker Desktop only enables certain experimental features after checking the status of those features online. We have a web service that publishes the status of these features so we can quickly disable broken features if we find a dangerous bug after release, to give us time to fix it and publish a new download. It caches the results of the lookup on disk at `%appdata%\Roaming\Docker\features-cache.json` so once it has successfully cached the results, Docker Desktop should still allow access to these features even if you disconnect from the internet or our web service is down.

We have tried to make this process as seamless/invisible as possible but obviously if it goes wrong that means it's hard to tell whats happened. We are discussing having a better error reporting system for the user or making this process more visible when it goes wrong.

So in summary, it's possible that you opened the settings window before it had a chance to contact the web server to enable it, and it's also possible the web server was down at the time, or maybe you were disconnected from the internet. You may have a 'failed to contact server' error in your log file. Sorry for the confusing behaviour!
